packages/router/test/int/PendleAdapter.spec.ts

```diff
@@ -0,0 +1,271 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import {
+  ERC20,
+  MarginlyRouter,
+  MarginlyRouter__factory,
+  PendleAdapter,
+  PendleAdapter__factory,
+} from '../../typechain-types';
+import { constructSwap, Dex, SWAP_ONE } from '../shared/utils';
+import { EthAddress } from '@marginly/common';
+import { formatUnits, keccak256, parseUnits } from 'ethers/lib/utils';
+import { BigNumber } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+
+// TODO remove me after all the merges are resolved
+function getAccountBalanceStorageSlot(account: EthAddress, tokenMappingSlot: string): string {
+  return keccak256('0x' + account.toString().slice(2).padStart(64, '0') + tokenMappingSlot);
+}
+
+// TODO remove me after all the merges are resolved
+export async function setTokenBalance(
+  tokenAddress: string,
+  balanceOfSlotAddress: string,
+  account: EthAddress,
+  newBalance: BigNumber
+) {
+  const balanceOfStorageSlot = getAccountBalanceStorageSlot(account, balanceOfSlotAddress);
+
+  await ethers.provider.send('hardhat_setStorageAt', [
+    tokenAddress,
+    balanceOfStorageSlot,
+    ethers.utils.hexlify(ethers.utils.zeroPad(newBalance.toHexString(), 32)),
+  ]);
+}
+
+// TODO remove me after all the merges are resolved
+export enum ArbMainnetERC20BalanceOfSlot {
+  WETH = '0000000000000000000000000000000000000000000000000000000000000033',
+  PTWEETH = '0000000000000000000000000000000000000000000000000000000000000000',
+}
+
+describe('Pendle swap pre maturity', () => {
```

Comment:
The block number in Hardhat fork config is not configured, so these tests will be incorrect in a few months. I suggest adding a fork config for Pendle as in [Pendle oracle PR](packages/periphery/hardhat-configs/fork-pendle-before-maturity.config.ts)